lib/mayaUsd/ufe/UsdTransform3dCommonAPI.cpp

```diff
@@ -27,167 +28,96 @@ namespace ufe {
+class CommonAPIRotateUndoableCmd : public UsdSetXformOpUndoableCommandBase<GfVec3f>
 {
 public:
-    UsdRotateUndoableCmd(const UsdSceneItem::Ptr& item, const UsdTimeCode& time)
-        : Ufe::RotateUndoableCommand(item->path())
-        , _time(time)
+    CommonAPIRotateUndoableCmd(const UsdSceneItem::Ptr& item, const UsdTimeCode& writeTime)
+        : UsdSetXformOpUndoableCommandBase(item->path(), writeTime)
         , _commonAPI(item->prim())
     {
-        GfVec3d                              t;
-        GfVec3f                              r, s, pvt;
-        UsdGeomXformCommonAPI::RotationOrder rotOrder;
-
-        if (!_commonAPI.GetXformVectorsByAccumulation(&t, &r, &s, &pvt, &rotOrder, time)) {
-            TF_FATAL_ERROR(
-                "Cannot read common API transform values for prim %s",
-                item->prim().GetPath().GetText());
-        }
-
-        _prevR = r;
-        _r = r;
     }
 
-    void undo() override
+    void setValue(const GfVec3f& v) override
     {
-        _commonAPI.SetRotate(_prevR, UsdGeomXformCommonAPI::RotationOrderXYZ, _time);
-    }
-    void redo() override
-    {
-        _commonAPI.SetRotate(_r, UsdGeomXformCommonAPI::RotationOrderXYZ, _time);
+        _commonAPI.SetRotate(v, UsdGeomXformCommonAPI::RotationOrderXYZ, writeTime());
     }
 
     // Executes the command by setting the rotation onto the transform op.
     bool set(double x, double y, double z) override
     {
-        _r = GfVec3f(x, y, z);
-        redo();
+        handleSet(GfVec3f(x, y, z));
         return true;
     }
 
 private:
-    const UsdTimeCode     _time;
     UsdGeomXformCommonAPI _commonAPI;
-    GfVec3f               _prevR, _r;
 };
 
-class UsdScaleUndoableCmd : public Ufe::ScaleUndoableCommand
+class CommonAPIScaleUndoableCmd : public UsdSetXformOpUndoableCommandBase<GfVec3f>
 {
 
 public:
-    UsdScaleUndoableCmd(const UsdSceneItem::Ptr& item, const UsdTimeCode& time)
-        : Ufe::ScaleUndoableCommand(item->path())
-        , _time(time)
+    CommonAPIScaleUndoableCmd(const UsdSceneItem::Ptr& item, const UsdTimeCode& writeTime)
+        : UsdSetXformOpUndoableCommandBase(item->path(), writeTime)
         , _commonAPI(item->prim())
     {
-        GfVec3d                              t;
-        GfVec3f                              r, s, pvt;
-        UsdGeomXformCommonAPI::RotationOrder rotOrder;
-
-        if (!_commonAPI.GetXformVectorsByAccumulation(&t, &r, &s, &pvt, &rotOrder, time)) {
-            TF_FATAL_ERROR(
-                "Cannot read common API transform values for prim %s",
-                item->prim().GetPath().GetText());
-        }
-
-        _prevS = s;
-        _s = s;
     }
 
-    void undo() override { _commonAPI.SetScale(_prevS, _time); }
-    void redo() override { _commonAPI.SetScale(_s, _time); }
+    void setValue(const GfVec3f& v) override { _commonAPI.SetScale(v, writeTime()); }
 
     // Executes the command by setting the rotation onto the transform op.
```

Comment:
@ppt-adsk typo:  "setting the rotation" ----> "setting the scale"

Reply:
Darned copy-paste...  Good catch!  Actually, this comment and the ones in the other classes were really low value, I just removed them locally.